Applied review comments by Dale Worley and Peter Saint-Andre (#73)
ietf-wg-spice/draft-ietf-spice-glue-id: draft-ietf-spice-glue-id.md

## Uniqueness and Namespacing

Each GLUE URN is globally unique identifier.
An organization can be identified by multiple GLUE URNs,
but each distinct GLUE URN can only refer to a single organization.
If multiple External Authorities have issued identifiers for the same organization
(e.g., the organization has both a Dun & Bradstreet number and a Private Enterprise Number),
then the same organization can be identified by GLUE URNs differentiated by the Authority Identifier.

It is assumed that most registered organizational entity identification schemes
already handle any namespacing necessary to achieve uniqueness as part of the External Identifier.
However, if collisions are possible within the set of possible external
identifiers for an Authority Identifier scheme, then further namespacing is
necessary at the GLUE URN level.

> Such namespacing MUST be done using the Authority Identifier.

Reviewer: ```suggestion
Such namespacing could be done by allocating multiple Authority Identifiers.
```

I don't think that your MUST is right.  You can either add more information to the authority identifier or more information to the external identifier, which is what the next sentence (the important one) is saying.

External Identifier is the representation of a number. It is specific to the
Authority Identifier whether the External Identifiers are case-insensitive or
case-sensitive. When they are case-insensitive, the canonical form is lowercase
and documents that specify External Identifiers MUST do so with lowercase letters.
Always using the canonical form of these URNs means that code performing comparisons
need not be aware of whether External Identifiers are case-sensitive or not;

> case-sensitive comparisons can always be performed.

Reviewer: This isn't quite right, because the first two parts of the URN are case-insensitive too "urn:glue:" or "URN:GlUe", where you cannot make the same requirement.

```suggestion
case-sensitive comparisons can always be performed
on the namespace specific string.
```

```suggestion
case-sensitive comparisons can always be performed
on the namespace specific string (that is, the Authority and External Identifiers).
```

While the original representation of some External Identifiers may contain
characters that are ignored, these MUST be omitted when used in GLUE URNs.
For instance, Dun & Bradstreet {{DUNS}} identifiers sometimes contain
dash characters that are ignored.
The DUNS numbers 12-345-6789 and 123456789 are considered to be equivalent.
The latter form without the ignored characters MUST be used
as the External Identifier in GLUE URNs.

Finally, should the original represenation of an External Identifier
contain the colon character, an alternative character,

> such as period, MUST be substituted for the colon when used in GLUE URNs.

Reviewer: It's not clear from this text where this is a requirement on those constructing an external identifier, or whether it is a requirement on the definition of a new GLUE URN type.  It has to be the latter.

So maybe instead:

```suggestion
Finally, documents that define a new Authority Identifier type
MUST NOT allow the representation of an External Identifier
to contain the colon character.
Specifications MUST define a substitute character,
such as period, that is used in place of a colon in External Identifiers.
```

The substitution performed when creating the GLUE URN
is specific to the Authority Identifier

> and is specified in the Transformation Rules in the IANA registration.

Reviewer: ```suggestion
Any substitution can be specified in the Transformation Rules
in the IANA registration for an Authority Identifier; see {{GLUE-Authority-Reg}}.
```

External Identifier is the representation of a number. It is specific to the
Authority Identifier whether the External Identifiers are case-insensitive or
case-sensitive. When they are case-insensitive, the canonical form is lowercase
and documents that specify External Identifiers MUST do so with lowercase letters.
Always using the canonical form of these URNs means that code performing comparisons
need not be aware of whether External Identifiers are case-sensitive or not;
case-sensitive comparisons can always be performed
on the namespace specific string.

While the original representation of some External Identifiers may contain
characters that are ignored, these MUST be omitted when used in GLUE URNs.
For instance, Dun & Bradstreet {{DUNS}} identifiers sometimes contain

> dash characters that are ignored.

Author: ```suggestion
hyphen characters that are ignored.
```

Organization:
: Dun & Bradstreet

Transformation Rules:

> : Delete dash characters.

Reviewer: Are these dashes, or hyphens?  (Or [HYPHEN-MINUS](https://unicodeplus.com/U+002D)).

Organization:
: Dun & Bradstreet

Transformation Rules:

> : Delete dash characters.

Author: ```suggestion
: Delete hyphen characters.
```
